compaction/compaction_manager.cc

```diff
@@ -1185,6 +1185,33 @@ future<> compaction_manager::perform_sstable_scrub_validate_mode(replica::table*
     return perform_task(seastar::make_shared<validate_sstables_compaction_task>(*this, t, std::move(all_sstables)));
 }
 
+future<> compaction_manager::maybe_wait_on_current_compaction(replica::table* t) {
+    auto sstables = boost::copy_range<std::vector<sstables::shared_sstable>>(
+            *(t->get_sstables()) |
+            boost::adaptors::filtered([](const sstables::shared_sstable& s) { return s->get_origin() == "memtable"; }));
+    sstables = get_non_compacting_sstables(sstables);
+    if (sstables.size() < t->schema()->max_compaction_threshold() || t->is_auto_compaction_disabled_by_user()) {
```

Comment:
The goal here is to prevent the node from dying if compaction can't keep up, not to satisfy the compaction strategy goals. Since the means are drastic (stopping writes) then the condition should be drastic too, like the total number of sstables being way too large (like 100,000/shard - even for LCS sstables that are 160 MB that gives 16 TB/shard, which is comfortably away from our usual ~1 TB/shard.

Note we'll need to increase LimitNOFile in scylla-server.service but that's outside the scope. 

@raphaelsc 160MB seems too small, let's increase it.

Reply:
We should look at number of runs instead. See that we'll filter in sstables with memtable origin only, so in LCS we'll only count SSTable runs in level 0. It's correct to count runs, so it doesn't work unexpectedly if memtable flush decides to create a SSTable run through segregator.

Reply:
@avikivity agreed about increasing default fragment size. we can use 1G as in ICS. So for 1TB, that's 1000 files. With storage density increasing, 160M is way too low

Reply:
We may indeed start facing lots of demotion as level capacity has fragment size as a factor, so levels will become underutilized, triggering demotion.

I think the plan is to kill 'level capacity' concept and instead use the dynamic level approach, where strategy is satisfied once L is fan_out times larger than L-1.
Then we can increase fragment size to 1G